src/odemis/driver/smaract.py

```diff
@@ -344,26 +346,38 @@ def __init__(self, name: str, role: str, locator: str, hwmodel: int=10001,
         else:
             self.core = SmarPodDLL()
             
-        # Not to be mistaken with axes which is a simple public view
-        self._axis_map = {}  # axis name -> axis number used by controller
+        # axis name (as seen by user) -> axis name (as used by hardware)
+        self._axis_u2hw: Dict[str, str] = {}
         axes_def = {}  # axis name -> Axis object
 
-        for axis_name, param in axes.items():
+        for axis_hw, param in axes.items():
+            if axis_hw not in ("x", "y", "z", "rx", "ry", "rz"):
+                raise ValueError(f"Axis {axis_hw} unknown")
+
+            # Name, as seen by the user (default is the same as the hardware one)
+            axis_name = param.get("name", axis_hw)
+            if axis_name in self._axis_u2hw:
+                raise ValueError(f"Axis {axis_name} used multiple times")
+            self._axis_u2hw[axis_name] = axis_hw
+
             try:
                 axis_range = param['range']
             except KeyError:
-                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
+                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_hw)
                 axis_range = (-1, 1)
 
             try:
                 axis_unit = param['unit']
             except KeyError:
-                logging.info("Axis %s has no unit. Assuming m", axis_name)
-                axis_unit = "m"
+                axis_unit = "rad" if axis_hw.startswith("r") else "m"
+                logging.info("Axis %s has no unit. Assuming %s", axis_unit, axis_hw)
```

Comment:
Shouldn't axis_ unit and axis_hw not be switched in this case?
```suggestion
                logging.info("Axis %s has no unit. Assuming %s", axis_hw, axis_unit)
```

Reply:
Good catch!